src/transformers/generation_tf_utils.py

```diff
@@ -2029,13 +2029,8 @@ def greedy_search_body_fn(generated, finished_sequences, next_tokens, current_po
                     decoder_hidden_states.append(outputs.hidden_states)
 
             # pre-process distribution
-            # TODO(pvp, joao, matt) - all the logits processors need to be adapted
-            # to be XLA compatible
-            input_ids = None
-            if not use_xla:
-                input_ids = tf.reshape(generated.concat(), (-1, batch_size))
-                input_ids = tf.transpose(input_ids[: current_pos[0]])
```

Comment:
no need to slice `input_ids`, as the updated `logits_processor` is expecting as input the fixed-length input